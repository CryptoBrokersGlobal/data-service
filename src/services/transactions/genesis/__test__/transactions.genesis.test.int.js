const createService = require('..');
const { createPgDriver } = require('../../../../db');

const TX_ID = '2DVtfgXjpMeFf2PQCqvwxAiaGbiDsxDjSdNQkc5JQ74eWxjWFYgwvqzC4dn7iB1AhuM32WxEiVi1SGijsBtYQwn8';
const TX_ID_2 = '2TsxPS216SsZJAiep7HrjZ3stHERVkeZWjMPFcvMotrdGpFa6UCCmoFiBGNizx83Ks8DnP3qdwtJ8WFcN9J4exa3';

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('Genesis transaction service', () => {
  describe('get', () => {
    it('fetches real tx', async done => {
      service
        .get(TX_ID)
        .run()
        .promise()
        .then(x => {
          expect(x).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
    it('returns null for unreal tx', async () => {
      const tx = await service
        .get('UNREAL')
        .run()
        .promise();

      expect(tx).toBe(null);
    });
  });

  describe('mget', () => {
    it('fetches real txs with nulls for unreal', async done => {
      service
        .mget([TX_ID, 'UNREAL', TX_ID_2])
        .run()
        .promise()
        .then(xs => {
          expect(xs).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
  });
});
