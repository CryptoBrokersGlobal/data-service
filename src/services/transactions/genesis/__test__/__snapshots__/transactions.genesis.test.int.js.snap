// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genesis transaction service get fetches real tx 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "amount": "99999995",
    "fee": "0",
    "height": 1,
    "id": "2DVtfgXjpMeFf2PQCqvwxAiaGbiDsxDjSdNQkc5JQ74eWxjWFYgwvqzC4dn7iB1AhuM32WxEiVi1SGijsBtYQwn8",
    "recipient": "3PAWwWa6GbwcJaFzwqXQN5KQm7H96Y7SHTQ",
    "signature": "2DVtfgXjpMeFf2PQCqvwxAiaGbiDsxDjSdNQkc5JQ74eWxjWFYgwvqzC4dn7iB1AhuM32WxEiVi1SGijsBtYQwn8",
    "timestamp": 2016-06-12T11:42:57.614Z,
    "type": 1,
  },
}
`;

exports[`Genesis transaction service mget fetches real txs with nulls for unreal 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "99999995",
        "fee": "0",
        "height": 1,
        "id": "2DVtfgXjpMeFf2PQCqvwxAiaGbiDsxDjSdNQkc5JQ74eWxjWFYgwvqzC4dn7iB1AhuM32WxEiVi1SGijsBtYQwn8",
        "recipient": "3PAWwWa6GbwcJaFzwqXQN5KQm7H96Y7SHTQ",
        "signature": "2DVtfgXjpMeFf2PQCqvwxAiaGbiDsxDjSdNQkc5JQ74eWxjWFYgwvqzC4dn7iB1AhuM32WxEiVi1SGijsBtYQwn8",
        "timestamp": 2016-06-12T11:42:57.614Z,
        "type": 1,
      },
    },
    null,
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "1",
        "fee": "0",
        "height": 1,
        "id": "2TsxPS216SsZJAiep7HrjZ3stHERVkeZWjMPFcvMotrdGpFa6UCCmoFiBGNizx83Ks8DnP3qdwtJ8WFcN9J4exa3",
        "recipient": "3P8JdJGYc7vaLu4UXUZc1iRLdzrkGtdCyJM",
        "signature": "2TsxPS216SsZJAiep7HrjZ3stHERVkeZWjMPFcvMotrdGpFa6UCCmoFiBGNizx83Ks8DnP3qdwtJ8WFcN9J4exa3",
        "timestamp": 2016-06-12T11:42:57.614Z,
        "type": 1,
      },
    },
  ],
}
`;
